src/main/dsl/Group.kt

```diff
@@ -1,5 +1,5 @@
 package dsl
 
-abstract class Group : Node(){
-
-}
\ No newline at end of file
+abstract class Group : Regex() {
```

Comment:
I'm not sure about making a Group extend a Regex. Won't that allow a bunch of nesting we don't want, like literals inside literals? I think we need stronger typechecking than what this provides.

Reply:
For most of the groups I believe the nesting is fine. For the nested literal thing I have a partial fix - since the type of `Regex::literal()` takes in a `() -> String`, it prevents nesting of literal blocks, but you can still get away with some weird syntax, like putting in a Group within a literal but still returning a String at the end. Moving forward we definitely want to have a better solution.